src/dependency-manager/spec/database-spec.ts

```diff
@@ -0,0 +1,37 @@
+import { IsOptional, IsString } from 'class-validator';
+import { JSONSchema } from 'class-validator-jsonschema';
+import { AnyOf, ExpressionOr } from './utils/json-schema-annotations';
+import { Slugs } from './utils/slugs';
+
+export class DatabaseSpec {
+  @IsOptional()
+  @JSONSchema({
+    type: 'string',
+    pattern: Slugs.ArchitectSlugValidator.source,
+    errorMessage: Slugs.ArchitectSlugDescription,
+    description: 'A specific service name which will override the database name specified in the component.',
```

Comment:
why would we need this? This seems extremely confusing for a new user on the first touch. I'm assuming it's for host overrides.

I don't know why a user would want to override the database name they specify in the component - I assume they would want to keep it named the same so they know what it is

Reply:
I don't remember the use case exactly, but @tjhiggins can chime in. He was the one who added that to services so I just tried to mirror it for now.